api/src/test/java/io/druid/data/input/impl/TimestampSpecTest.java

```diff
@@ -37,6 +39,29 @@ public void testExtractTimestamp() throws Exception
   }
 
   @Test
+  public void testExtractTimestampWithTimezone() throws Exception
+  {
+    final DateTimeZone saved = DateTimeZone.getDefault();
+    final DateTimeZone dataTimezone = DateTimeZone.forID("Asia/Shanghai");
+    try {
+      DateTimeZone.setDefault(DateTimeZone.UTC);
+
+      DateTime now = new DateTime();
+      String nowShanghai = DateTimeFormat.forPattern("yyyy-MM-dd HH:mm:ss.SSS").print(now.withZone(dataTimezone));
```

Comment:
Actually it's deterministic since I printed the DateTime in millisecond precision, and the extracted timestamp should represent the same instant as `now`. Please correct me if I missed something.